crypto/merkle.go

```diff
@@ -70,45 +92,39 @@ func CalculateLeaves(fileSize uint64) uint64 {
 	return numSegments
 }
 
-// ReaderMerkleRoot returns the Merkle root of a reader.
-func ReaderMerkleRoot(r io.Reader) (h Hash, err error) {
-	root, err := merkletree.ReaderRoot(r, NewHash(), SegmentSize)
-	if err != nil {
-		return
+// MerkleRoot returns the Merkle root of a reader.
+func MerkleRoot(b []byte) Hash {
```

Comment:
comment is wrong.

I think I would prefer to continue using `merkletree.ReaderRoot` internally, because it lets us offload critical logic to the merkletree package, rather than reimplementing it here.